### DISCUSSION TITLE
gcf5.0-Using a new raw file as the input model of transportFoam (Ketton)

### DISCUSSION THREAD

**Discussion by ZZZYunpeng** [GeoChemFoam/GeoChemFoam#21] [Q&A] [answered]
Dear inventors and net friends,

I have a question about **how to use my raw file as the input model of transportFoam** (i.e., [the tutorial 01 in users wiki](https://github.com/GeoChemFoam/GeoChemFoam/wiki/Test-Case-01----Species-transport)).

I made a new raw file by using Porepy and Numpy. Then I uploaded it into Ketton which is the tutorial of flow in a Micro-CT image.
Firstly, I keyed command './creatMesh.sh'. The results showed that everything was ok.
Secondly, I keyed command './runSnappyHexMesh.sh'. There were no errors reported during this process. But the 'inlet patch' disappeared, which existed in the first step. The snappyHexmesh generated did not seem quite right.
Finally, I followed the tutorial to complete my simulation. No errors were reported throughout the process. But there is nothing in the results.

**So I would like to ask the reasons and how you create a Micro-CT raw file.** 
This is my simulation [data](https://github.com/ZZZYunpeng/Yunpeng_data).

Best regards,
Yunpeng

**Comment by GeoChemFoam** [comment]
Hi,

Can you check that you have conformed to the image requirements discussed in this tutorial? Some of the later steps need to be updated for the latest version of GCF, but it should give you some hints.
https://github.com/GeoChemFoam/GeoChemFoam/wiki/Solving-permeability-on-an-open-access-binary-image


Best wishes,

Hannah


From: ZZZYunpeng ***@***.***>
Sent: Monday, September 25, 2023 7:35 AM
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GeoChemFoam/GeoChemFoam] gcf5.0-Using a new raw file as the input model of transportFoam (Ketton) (Discussion #21)

****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Dear inventors and net friends,

I have a question about how to use my raw file as the input model of transportFoam (i.e., the tutorial 01 in users wiki<https://github.com/GeoChemFoam/GeoChemFoam/wiki/Test-Case-01----Species-transport>).

I made a new raw file by using Porepy and Numpy. Then I uploaded it into Ketton which is the tutorial of flow in a Micro-CT image.
Firstly, I keyed command './creatMesh.sh'. The results showed that everything was ok.
Secondly, I keyed command './runSnappyHexMesh.sh'. There were no errors reported during this process. But the 'inlet patch' disappeared, which existed in the first step. The snappyHexmesh generated did not seem quite right.
Finally, I followed the tutorial to complete my simulation. No errors were reported throughout the process. But there is nothing in the results.

So I would like to ask the reasons and how you create a Micro-CT raw file.
This is my simulation data<https://github.com/ZZZYunpeng/Yunpeng_data>.

Best regards,
Yunpeng

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/21>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFSZWTZKESTK3XTTS73X4EQX5ANCNFSM6AAAAAA5FSJ5GE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Reply by julienmaes** [reply, marked as answer]
If the mesh looks OK except that the inlet has disappeared, this probably means that snappyHexMesh thinks that your image is not connected. SnappyHexMesh is not very good at that and struggle to identify connection when throats are 1 or 2 voxels. Try increasing your mesh size if you can.

Alternatively use DBS, for example the way it is used in the heatTransfer tutorial
